gensim/models/sent2vec.py

```diff
@@ -0,0 +1,788 @@
+            >>> model = Sent2Vec(min_count=1)
+            >>> model.build_vocab(sentences)
+            >>> model.train(sentences, total_examples=model.corpus_count, epochs=model.epochs)
+
+        """
+        if(corpus_file is not None):
+            raise RuntimeError(
+                "Aruguement corpus_file is not None.")
+        if FAST_VERSION < 0:
+            return do_train_job_slow(self, sentences)
+        return super(Sent2Vec, self).train(
+            sentences, total_examples=total_examples, total_words=total_words,
+            epochs=epochs, start_alpha=start_alpha, end_alpha=end_alpha, word_count=word_count,
+            queue_factor=queue_factor, report_delay=report_delay, compute_loss=compute_loss,
+            callbacks=callbacks, corpus_file=corpus_file)
+
+    def __getitem__(self, sentence):
+        """Get sentence vector for an input sentence.
+
+        Parameters
+        ----------
+        sentence : list of str
+            List of words.
+
+        Returns
+        -------
+        numpy.ndarray
+            Sentence vector for input sentence.
+
+        """
+        ntokens_temp, words = self.vocabulary.get_line(sentence)
+        sent_vec = np.zeros(self.vector_size)
+        line = self.vocabulary.add_word_ngrams(context=words, n=self.word_ngrams)
+        if len(line) > 0:
+            sent_vec = np.mean(self.wi[line], axis=0)
+        return sent_vec
+
+    @classmethod
+    def load(cls, *args, **kwargs):
+        return super(BaseWordEmbeddingsModel, cls).load(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        kwargs['ignore'] = kwargs.get('ignore', ['hidden', 'grad'])
+        return super(Sent2Vec, self).save(*args, **kwargs)
+
+    def similarity(self, sent1, sent2):
+        """Function to compute cosine similarity between two sentences.
+
+        Parameters
+        ----------
+        sent1 : list of str
+            List of words.
+        sent2 : list of str
+            List of words.
+
+        Returns
+        -------
+        float
+            Cosine similarity score between two sentence vectors.
+
+        """
+        return dot(matutils.unitvec(self[sent1]), matutils.unitvec(self[sent2]))
+
+
+class TorontoCorpus(object):
+    """Iterate over sentences from the Toronto Book Corpus."""
+    def __init__(self, dirname):
+        """
+
+        Parameters
+        ----------
+        dirname : str
+            Name of the directory where the dataset is located.
+
+        """
+        self.dirname = dirname
+
+    def __iter__(self):
+        for fname in os.listdir(self.dirname):
+            fname = os.path.join(self.dirname, fname)
+            if not os.path.isfile(fname) or ".txt" not in fname:
+                continue
+            sentence = None
+            for line in utils.smart_open(fname):
+                if line not in ['\n', '\r\n']:
+                    sentence = list(tokenize(line))
+                if sentence:
+                    yield sentence
```

Comment:
if line is  a new line chars then previous sentence will be yielded twice.

Reply:
hm, I guess that's a bug